File: app/src/main/java/wannabit/io/cosmostaion/dialog/CurrencySetDialog.java
package wannabit.io.cosmostaion.dialog;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import wannabit.io.cosmostaion.R;
import wannabit.io.cosmostaion.base.BaseConstant;
import wannabit.io.cosmostaion.utils.SimpleDividerItemDecoration;

public class CurrencySetDialog extends DialogFragment {

    public final static String CURRENCY_SET_BUNDLE_KEY = "currency";

    private RecyclerView mRecyclerView;
    private TextView mDialogTitle;
    private CurrencySetAdapter mCurrencySetAdapter;

    public static CurrencySetDialog newInstance(Bundle bundle) {
        CurrencySetDialog frag = new CurrencySetDialog();
        frag.setArguments(bundle);
        return frag;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        getDialog().getWindow().setBackgroundDrawableResource(R.drawable.layout_trans_with_border);
        View view = LayoutInflater.from(getActivity()).inflate(R.layout.dialog_template_choice, null);
        mRecyclerView = view.findViewById(R.id.recycler);
        mDialogTitle = view.findViewById(R.id.dialog_title);
        mDialogTitle.setVisibility(View.GONE);
        mRecyclerView.setLayoutManager(new LinearLayoutManager(getContext(), LinearLayoutManager.VERTICAL, false));
        mRecyclerView.setHasFixedSize(true);
        mCurrencySetAdapter = new CurrencySetAdapter();

        mRecyclerView.addItemDecoration(new SimpleDividerItemDecoration(getActivity()));
        mRecyclerView.setAdapter(mCurrencySetAdapter);
        return view;
    }

    private class CurrencySetAdapter extends RecyclerView.Adapter<CurrencySetAdapter.CurrencyHolder> {

        @NonNull
        @Override
        public CurrencySetAdapter.CurrencyHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
            return new CurrencySetAdapter.CurrencyHolder(getLayoutInflater().inflate(R.layout.item_dialog_currency_set, viewGroup, false));
        }

        @Override
        public void onBindViewHolder(@NonNull CurrencySetAdapter.CurrencyHolder viewHolder, int position) {
            onCurrencyItem(viewHolder, position);
        }

        private void onCurrencyItem(CurrencyHolder holder, int position) {
            String[] mUnitList = getResources().getStringArray(R.array.currency_unit_array);

            holder.currencyName.setText(mUnitList[position]);

            holder.rootLayer.setOnClickListener(v -> {
                Bundle result = new Bundle();
                result.putInt(BaseConstant.POSITION, position);
                getParentFragmentManager().setFragmentResult(CurrencySetDialog.CURRENCY_SET_BUNDLE_KEY, result);
                dismiss();
            });
        }

        @Override
        public int getItemCount() {
            return getResources().getStringArray(R.array.currency_unit_array).length;
        }

        public class CurrencyHolder extends RecyclerView.ViewHolder {
            LinearLayout rootLayer;
            TextView currencyName;

            public CurrencyHolder(@NonNull View itemView) {
                super(itemView);
                rootLayer = itemView.findViewById(R.id.rootLayer);
                currencyName = itemView.findViewById(R.id.currencyName);
            }
        }
    }
}

File: app/src/main/java/wannabit/io/cosmostaion/dialog/CurrencySetDialog.kt
package wannabit.io.cosmostaion.dialog

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.DialogFragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import wannabit.io.cosmostaion.R
import wannabit.io.cosmostaion.base.BaseConstant
import wannabit.io.cosmostaion.databinding.DialogTemplateChoiceBinding
import wannabit.io.cosmostaion.databinding.ItemDialogCurrencySetBinding
import wannabit.io.cosmostaion.dialog.CurrencySetDialog.CurrencySetAdapter.CurrencyHolder
import wannabit.io.cosmostaion.utils.SimpleDividerItemDecoration

class CurrencySetDialog : DialogFragment() {
    private var dialogTemplateChoiceBinding: DialogTemplateChoiceBinding? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        dialog!!.window!!.setBackgroundDrawableResource(R.drawable.layout_trans_with_border)
        dialogTemplateChoiceBinding =
            DialogTemplateChoiceBinding.inflate(inflater, container, false)
        dialogTemplateChoiceBinding!!.dialogTitle.visibility = View.GONE
        dialogTemplateChoiceBinding!!.recycler.layoutManager =
            LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false)
        dialogTemplateChoiceBinding!!.recycler.setHasFixedSize(true)
        dialogTemplateChoiceBinding!!.recycler.addItemDecoration(
            SimpleDividerItemDecoration(
                activity
            )
        )
        dialogTemplateChoiceBinding!!.recycler.adapter = CurrencySetAdapter()
        return dialogTemplateChoiceBinding!!.root
    }

    private inner class CurrencySetAdapter : RecyclerView.Adapter<CurrencyHolder>() {
        override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): CurrencyHolder {
            return CurrencyHolder(
                ItemDialogCurrencySetBinding.inflate(
                    layoutInflater
                )
            )
        }

        override fun onBindViewHolder(viewHolder: CurrencyHolder, position: Int) {
            onCurrencyItem(viewHolder, position)
        }

        private fun onCurrencyItem(holder: CurrencyHolder, position: Int) {
            val mUnitList = resources.getStringArray(R.array.currency_unit_array)
            holder.itemDialogCurrencySetBinding.currencyName.text = mUnitList[position]
            holder.itemDialogCurrencySetBinding.rootLayer.setOnClickListener {
                val result = Bundle()
                result.putInt(BaseConstant.POSITION, position)
                parentFragmentManager.setFragmentResult(CURRENCY_SET_BUNDLE_KEY, result)
                dismiss()
            }
        }

        override fun getItemCount(): Int {
            return resources.getStringArray(R.array.currency_unit_array).size
        }

        inner class CurrencyHolder(val itemDialogCurrencySetBinding: ItemDialogCurrencySetBinding) :
            RecyclerView.ViewHolder(
                itemDialogCurrencySetBinding.root
            )
    }

    override fun onDestroyView() {
        super.onDestroyView()
        dialogTemplateChoiceBinding = null
    }

    companion object {
        const val CURRENCY_SET_BUNDLE_KEY = "currency"

        @JvmStatic
        fun newInstance(bundle: Bundle?): CurrencySetDialog {
            val frag = CurrencySetDialog()
            frag.arguments = bundle
            return frag
        }
    }
}